satellite/gracefulexit/endpoint.go

```diff
@@ -843,6 +673,31 @@ func (endpoint *Endpoint) handleFailed(ctx context.Context, pending *pendingMap,
 	return nil
 }
 
+func (endpoint *Endpoint) handleFinished(ctx context.Context, stream processStream, exitStatusRequest *overlay.ExitStatusRequest, failedReason pb.ExitFailed_Reason) error {
+	finishedMsg, err := endpoint.getFinishedMessage(ctx, endpoint.signer, exitStatusRequest.NodeID, exitStatusRequest.ExitFinishedAt, exitStatusRequest.ExitSuccess, failedReason)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	_, err = endpoint.overlaydb.UpdateExitStatus(ctx, exitStatusRequest)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	err = stream.Send(finishedMsg)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	// remove remaining items from the queue after notifying nodes about their exit status
+	err = endpoint.db.DeleteTransferQueueItems(ctx, exitStatusRequest.NodeID)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	return nil
+}
+
 func (endpoint *Endpoint) getFinishedMessage(ctx context.Context, signer signing.Signer, nodeID storj.NodeID, finishedAt time.Time, success bool, reason pb.ExitFailed_Reason) (message *pb.SatelliteMessage, err error) {
```

Comment:
nit - we should have caught this before, but since it is not related to this PR directly, feel free to ignore. `getFinishedMessage` does not need to take `signer signing.Signer` as an argument because it can be accessed via `endpoint.signer`.